server/src/main/java/org/opensearch/index/remote/RemoteStorePathStrategy.java

```diff
@@ -77,19 +77,13 @@ public BlobPath generatePath(PathInput pathInput) {
      */
     @PublicApi(since = "2.14.0")
     @ExperimentalApi
-    public static class PathInput {
+    public static class BasePathInput {
```

Comment:
This will cause a breaking change

Reply:
We should model the arguments that would be fed to hash function as variable arguments.

Reply:
@ashking94 Since arguments have different data types and we are doing some validations on that, it would become a bit unclean to convert data types. I feel this is a better way where we have a base path and then we can create custom path objects based on requirement on top of that.

Reply:
@Bukhtawar  Will change the name back to maintain compatibility

Reply:
Additionally, all these are marked with ExperimentalAPI flag.